pkg/kubelet/kubelet_node_status.go

```diff
@@ -579,7 +580,17 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	}
 
 	node, changed := kl.updateNode(ctx, originalNode)
-	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency
+
+	passStatusReportInterval := !kl.lastStatusReportTime.IsZero() && kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency+kl.delayAfterNodeStatusChange
```

Comment:
code lgtm. This is a good improvement.

Can you please add tests? If full function test is hard, maybe make it into the utility method that can be unit tested.

Reply:
Thanks for the reivew!
Let me see what I can do for adding tests.